spyderlib/plugins/shortcuts.py

```diff
@@ -7,179 +7,504 @@
+        doc.setHtml(options.text)
+        doc.setTextWidth(options.rect.width())
+
+        return QSize(doc.idealWidth(), doc.size().height())
+
+
+class ShortcutFinder(QLineEdit):
+    """Textbox for filtering listed shortcuts in the table."""
+    def __init__(self, parent, callback=None):
+        super(ShortcutFinder, self).__init__(parent)
+
+        self._parent = parent
+        regex = QRegExp(VALID_FINDER_CHARS + "{100}")
+        self.setValidator(QRegExpValidator(regex))
+
+        # Signals
+        if callback:
+            self.textChanged.connect(callback)
+
+    def set_text(self, text):
+        """Set the filter text"""
+        text = text.strip()
+        new_text = self.text() + text
+        self.setText(new_text)
+
+    def keyPressEvent(self, event):
+        """Qt Override"""
+        key = event.key()
+        if key in [Qt.Key_Up]:
+            self._parent.previous_row()
+        elif key in [Qt.Key_Down]:
+            self._parent.next_row()
+        elif key in [Qt.Key_Enter, Qt.Key_Return]:
+            self._parent.show_editor()
+        else:
+            super(ShortcutFinder, self).keyPressEvent(event)
+
+
+class ShortcutEditor(QDialog):
+    """A dialog for entering key sequences."""
+    def __init__(self, parent, context, name, sequence, shortcuts):
+        super(ShortcutEditor, self).__init__(parent)
+        self.context = context
+        self.npressed = 0
+        self.keys = set()
+        self.key_modifiers = set()
+        self.key_non_modifiers = list()
+        self.key_text = list()
+        self.sequence = sequence
+        self.new_sequence = None
+        self.edit_state = True
+        self.shortcuts = shortcuts
+
+        # Widgets
+        self.label_info = QLabel(_("Press the new shortcut and select 'Ok': \n"
+                                   "(Press 'Tab' once to switch focus between "
+                                   "shortcut entry and buttons)"))
+        self.label_current_sequence = QLabel(_("Current shortcut:"))
+        self.text_current_sequence = QLabel(sequence)
+        self.label_new_sequence = QLabel(_("New shortcut:"))
+        self.text_new_sequence = CustomLineEdit(self)
+        self.helper_button = HelperToolButton()
+
+        bbox = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        self.button_ok = bbox.button(QDialogButtonBox.Ok)
+        self.button_cancel = bbox.button(QDialogButtonBox.Cancel)
+
+        # Setup widgets
+        self.setWindowTitle(_('Shortcut: {0}').format(name))
+        self.button_ok.setFocusPolicy(Qt.NoFocus)
+        self.button_ok.setEnabled(False)
+        self.button_cancel.setFocusPolicy(Qt.NoFocus)
+        self.helper_button.setToolTip('')
+        self.helper_button.setFocusPolicy(Qt.NoFocus)
+        style = """
+            QToolButton {
+              margin:1px;
+              border: 0px solid grey;
+              padding:0px;
+              border-radius: 0px;
+            }"""

```

Comment:
We have a qss stylesheet for Mac already. But let's leave that for another PR :-)